src/main/java/io/cryostat/net/web/http/api/v2/RulesPostHandler.java

```diff
@@ -0,0 +1,208 @@
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.net.web.http.api.v2;
+
+import java.io.IOException;
+import java.util.function.Function;
+
+import javax.inject.Inject;
+
+import io.cryostat.core.log.Logger;
+import io.cryostat.net.AuthManager;
+import io.cryostat.net.web.http.HttpMimeType;
+import io.cryostat.net.web.http.api.ApiVersion;
+import io.cryostat.rules.Rule;
+import io.cryostat.rules.RuleRegistry;
+
+import com.google.gson.Gson;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+
+class RulesPostHandler extends AbstractV2RequestHandler<String> {
+
+    static final String PATH = "rules";
+
+    private final RuleRegistry ruleRegistry;
+    private final Logger logger;
+
+    @Inject
+    RulesPostHandler(AuthManager auth, RuleRegistry ruleRegistry, Gson gson, Logger logger) {
+        super(auth, gson);
+        this.ruleRegistry = ruleRegistry;
+        this.logger = logger;
+    }
+
+    @Override
+    public boolean requiresAuthentication() {
+        return true;
+    }
+
+    @Override
+    public ApiVersion apiVersion() {
+        return ApiVersion.V2;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.POST;
+    }
+
+    @Override
+    public String path() {
+        return basePath() + PATH;
+    }
+
+    @Override
+    public HttpMimeType mimeType() {
+        return HttpMimeType.PLAINTEXT;
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    public boolean isOrdered() {
+        return true;
+    }
+
+    @Override
+    public IntermediateResponse<String> handle(RequestParameters params) throws ApiException {
+        Rule rule;
+        String rawMime = params.getHeaders().get(HttpHeaders.CONTENT_TYPE);
+        if (rawMime == null) {
+            throw new ApiException(415, "Bad content type: null");
+        }
+        String firstMime = rawMime.split(";")[0];
+        HttpMimeType mime = HttpMimeType.fromString(firstMime);
+        if (mime == null) {
+            throw new ApiException(415, "Bad content type: " + rawMime);
+        }
+        switch (mime) {
+            case MULTIPART_FORM:
```

Comment:
What's the purpose of this empty case?

Reply:
It's a fall-through. Both multipart forms and urlencoded forms can be handled the same.